frontend/src/components/App/Settings/SettingsCluster.tsx

```diff
@@ -59,10 +59,23 @@ export default function SettingsCluster() {
   const history = useHistory();
   const dispatch = useDispatch();
 
-  const handleUpdateClusterName = () => {
+  const clusterInfo = (clusterConf && clusterConf[cluster || '']) || null;
+  const source = clusterInfo?.meta_data?.source || '';
+
+  const handleUpdateClusterName = (source: string) => {
     try {
-      console.log('came here', newClusterName); // Call the backend function to update the cluster name
       storeNewClusterName(newClusterName);
+      renameCluster(cluster || '', newClusterName, source)
+        .then(config => {
+          dispatch(setConfig(config));
+          history.push('/');
+          window.location.reload();
+        })
+        .catch((err: Error) => {
+          if (err.message === 'Not Found') {
+            // TODO: create notification with error message
```

Comment:
Why is this a TODO?? Do you wish to do it in a different PR??

Reply:
I think this is not part of this PR, but in scope of notifications. Please refer to https://github.com/headlamp-k8s/headlamp/pull/1974/commits/c5f310e551daa7e4eac6dda5dfebf8018afbe3ac#diff-360436c42cc1c1880e31a8c726c2dc1cd6e52ed471da9443e9956ac8f7210d5aL79